standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -724,6 +724,9 @@ public enum ConfVars {
     DATANUCLEUS_AUTOSTART("datanucleus.autoStartMechanismMode",
         "datanucleus.autoStartMechanismMode", "ignored", new StringSetValidator("ignored"),
         "Autostart mechanism for datanucleus.  Currently ignored is the only option supported."),
+    DATANUCLEUS_CACHE_LEVEL1_TYPE("datanucleus.cache.level1.type",
+        "datanucleus.cache.level1.type", "soft", new StringSetValidator("none", "soft", "weak", "strong"),
+        "Stores instances within a PersistenceManager"),
```

Comment:
Do we need a similar change for `DATANUCLEUS_CACHE_LEVEL2_TYPE` ?

Reply:
it's already been here: https://github.com/apache/hive/blob/master/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java#L729

Reply:
I would like to leave it as it is, there are multiple implementations for this level cache, some are DataNucleus-provided, others are external cache, https://www.datanucleus.org/products/accessplatform_4_2/extensions/level2_cache.html